File: TP3/punto1.py
import numpy as np
import matplotlib.pyplot as plt

# 1. Generar 100 datos aleatorios (Media=150, Desvío=20)
np.random.seed(22) # Para que los resultados sean replicables
#*""" 
# np.random.normal, genera elementos que sigan la distribucion normal
# loc es la media a la qe queremos apuntar
# scale es el desvio estandar respecto al promedio
# size es la cantidad de elementos de la muestra
#  """
datos = np.random.normal(loc=150, scale=20, size=100)

# 2. Cálculos estadísticos
#np.mean suma los valores generados y luego los divide por la cantidad de valores
media = np.mean(datos)
#np.std que tan lejos esta cada dato del promedio
desvio = np.std(datos)
#np.var desvio estandar al cuadrado (se usa para analisis)

media = np.mean(datos)
desvio = np.std(datos)
varianza = np.var(datos)



print(f"Media: {media:.2f}")
print(f"Desvío Estándar: {desvio:.2f}")
print(f"Varianza: {varianza:.2f}")

plt.hist(datos, bins=20, color='green', edgecolor='black')
plt.title('Distribución de Pesos de Manzanas')
plt.xlabel('Gramos')
plt.ylabel('Frecuencia')
plt.show()

File: TP3/punto1.jl
# Si es la primera vez usando el lenguaje, se deben instalar los siguientes modulos por la consola de julia
# import Pkg; Pkg.add("Distributions")
# import Pkg; Pkg.add("StatsPlots")

using Distributions
using Statistics
using StatsPlots

# 1. Definir la distribución de pesos (en gramos)
distribucionNormal = Normal(150, 20)
datos = rand(distribucionNormal, 100)

# 2. Cálculos solicitados
println("Media del peso: ", mean(datos), " g")
println("Desvío estándar: ", std(datos), " g")
println("Varianza: ", var(datos))

# 3. Gráfico
histogram(datos, 
    bins=15, 
    normalize=:pdf, 
    fillcolor=:green,
    fillalpha=0.4, 
    label="Pesos reales (Muestra)",
    title="Distribución del Peso de las Manzanas",
    xlabel="Peso (gramos)",
    ylabel="Densidad")

# Agregamos la curva teórica perfecta
plot!(distribucionNormal, 
    linecolor=:blue, 
    lw=3, 
    label="Distribución Teórica")